### PR TITLE
adds glow orbs

### PR DESCRIPTION
**Pull request by ghost**
Adds glow orbs, something simple that wizards can buy, or can be found through xenoarch at house digsites

- Light up the place
- Can be commanded to follow you just by touching them gently (help intent)
- Can be turned off and pocketed by shaking them (Any intent but help)
- Will stop following you by also being touched gently
- If hit by lightning or taser fire, overload and blind everyone in the vicinity
- Keep out of reach of children and clowns

glow_stone

(The ingame image has some opacity to it, exporting it didn't keep this however)

### CODE REVIEW

**Comment by ghost** [comment]
>wizards can buy

does it break when it's overloaded? does it also blind wizards?
Seems kinda lame since he already has a blind spell

**Comment by ghost** [comment]
Literally so there's simpler, more mundane shit the wizards can buy

File: code/game/gamemodes/wizard/artefact.dm [hunk 1]
```diff
@@ -136,4 +136,66 @@
...
+		if (clumsy_check(user) && prob(50))
+			to_chat(user, "<span class = 'notice'>Ooh, shiny</span>")
+			failure()
+			return
+		else if(prob(65))
+			to_chat(user, "<span class = 'notice'>You find what appears to be an on button, and press it.</span>")
+			activate()
+		else
+			if(prob(5))
+				visible_message("<span class = 'warning'>\The [src] ticks [pick("ominously","forebodingly", "harshly")]</span>")
```
Comment (Intigracy) [inline code/game/gamemodes/wizard/artefact.dm]: period at end

File: code/game/gamemodes/wizard/artefact.dm [hunk 2]
```diff
@@ -136,4 +136,66 @@
...
+				if(prob(50))
```
Comment (Intigracy) [inline code/game/gamemodes/wizard/artefact.dm]: There's a 1.75% chance (35/100*5/100) of them even getting this far, I think you can remove this one.

File: code/game/gamemodes/wizard/artefact.dm [hunk 3]
```diff
@@ -136,4 +136,66 @@
...
+	if(activating)
+		to_chat(user, "<span class = 'warning'>\The [src] hums with energy as it begins to glow brighter.</span>")
+		return
+	if(iswizard(user) || isapprentice(user))
+		to_chat(user, "<span class = 'notice'>You prime the glow-stone, it will transform in [prime_time/10] seconds.</span>")
+		activate()
+		return
+	else
+		if (clumsy_check(user) && prob(50))
+			to_chat(user, "<span class = 'notice'>Ooh, shiny</span>")
```
Comment (Wizardcrying) [inline code/game/gamemodes/wizard/artefact.dm]: Period

File: code/game/gamemodes/wizard/artefact.dm [hunk 4]
```diff
@@ -136,4 +136,66 @@
...
+		flick("glow_stone_activate", src)
+		spawn(10)
+			new/mob/living/simple_animal/hostile/glow_orb(get_turf(src))
+			qdel(src)
+
+/obj/item/weapon/glow_orb/proc/failure()
+	visible_message("<span class = 'notice'>\The [src] begins to glow increasingly in a brilliant manner...</span>")
+	crit_failure = 1
+	spawn(1 SECONDS)
+		visible_message("<span class = 'warning>...and vibrate violently</span>")
```
Comment (Wizardcrying) [inline code/game/gamemodes/wizard/artefact.dm]: Ending punctuation.

File: code/modules/mob/living/simple_animal/hostile/glow_orb.dm [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+If touched by somebody on help intent, will attempt to follow them until they move out of view.
+
+If attacked or hit by anything, powers down back into /obj/item/weapon/glow_orb
+
+If hit by lightning, overpowers and explodes like a flashbang, blinding everyone in immediate vicinity
+*/
+
+/mob/living/simple_animal/hostile/glow_orb
+	name = "strange glowing orb"
+	desc = "A hovering glowing orb, drifting to and fro lazily under its own power."
```
Comment (Wizardcrying) [inline code/modules/mob/living/simple_animal/hostile/glow_orb.dm]: `to-and-fro`.

File: code/modules/mob/living/simple_animal/hostile/glow_orb.dm [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+	return
+
+/mob/living/simple_animal/hostile/glow_orb/attack_hand(mob/living/carbon/human/M)
+	if(M.a_intent == I_HELP)
+		if(M != following)
+			to_chat(M, "<span class = 'notice'>\The [src] bobs excitedly, and glides towards you.</span>")
+			following = M
+			set_light(4, 1, "#68E8FF")
+		else if(M == following)
+			to_chat(M, "<span class = 'notice'>\The [src] droops idly as it maintains position</span>")
```
Comment (Wizardcrying) [inline code/modules/mob/living/simple_animal/hostile/glow_orb.dm]: Period.

File: code/game/gamemodes/wizard/artefact.dm (hunk 2 above)
Comment (ghost) [reply code/game/gamemodes/wizard/artefact.dm]: critical failure's always worth keeping in

File: code/modules/mob/living/simple_animal/hostile/glow_orb.dm [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+		playsound(get_turf(src), 'sound/weapons/orb_deactivate.ogg', 50,1)
+		set_light(0)
+		new/obj/item/weapon/glow_orb(get_turf(src))
+		qdel(src)
+		return
+
+/mob/living/simple_animal/hostile/glow_orb/bullet_act(var/obj/item/projectile/P)
+	if(istype(P, /obj/item/projectile/beam/lightning/spell) || istype(P, /obj/item/projectile/energy/electrode))
+		if(!unstable)
+			visible_message("<span class = 'warning'>\The [src]'s light grows greater in intensity, and begins to shake uncontrollably</span>")
```
Comment (Wizardcrying) [inline code/modules/mob/living/simple_animal/hostile/glow_orb.dm]: Period.

File: code/game/gamemodes/wizard/artefact.dm (hunk 2 above)
Comment (Intigracy) [reply code/game/gamemodes/wizard/artefact.dm]: I meant keep the failure, remove the prob(50)

Comment (ghost) [reply code/game/gamemodes/wizard/artefact.dm]: but why?

Could be that it clicks, you go 'oh fuck', and throw it and it does nothing, it's only a minor failure not a critical failure

Comment (Intigracy) [reply code/game/gamemodes/wizard/artefact.dm]: 🤷‍♂️

**Comment by Blithering** [comment]
The trouble at present is anyone choosing this is giving up points that could be used for mutate, mind transfer, etc. It would be neat if the spellbook had a separate set of cantrips that could be bought with a different spell point set, with the lesser used utility spells like knock or smoke moved to that section

**Comment by Arthurdentist** [comment]
I wouldn't consider knock a lesser used spell, its very powerful and I don't think it ought to be discounted.

**Comment by Blithering** [comment]
People do tend to pass it up in favour of jaunt or blink, as it allows them to access secure areas just as easily, and also get out of trouble for the same price

**Comment by Arthurdentist** [comment]
that probably says far more about jaunt then knock. Knock is extremely powerful and great for stealth wizzies, doesn't need to be discounted.

**Pull request merged**